TelemetryManager/Src/TMCircularBuffer.cpp

```diff
@@ -1,30 +1,56 @@
 
-#include "CircularBuffer.hpp"
+#include "TMCircularBuffer.hpp"
 
-TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(buf, size) {
+TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(*buf, size) {
     // Constructor
 }
 
-TMCircularBuffer::TMCircularBuffer(CircularBuffer buf): CircularBuffer(buf) {
+TMCircularBuffer::TMCircularBuffer(CircularBuffer* buf): CircularBuffer(*buf) {
     // Constructor
 }
 
 TMCircularBuffer::~TMCircularBuffer() {
     // Destructor
-}
+    delete[] buf;
+    buf = nullptr;
```

Comment:
The deconstruction to the `buf_` shouldn't be handled by this class since the original buffer was passed by a pointer. If you deconstruct it here, the outer buffer pointer will also be deconstructed. There is also a good chance that the outer buffer isn't even dynamically allocated, in that case, deconstruct them here doesn't make any sense.